<!--
	Launch for for USB momo camera capture node.
	Requirement: run camera calibration and LDC look-up-table generation tool which are located under <mono_cature>/scripts
	   	for generating camera_info .yaml file and undistortion/rectification look-up table files for LDC hardware accelerator
-->

<launch>
	<!-- string for camera model -->
	<arg name="model_str"             default="C920"/>
	<!-- to find your device name, use ls /dev/video* and look for the name begin with video -->
	<!-- typically camera recoginized as "/dev/video1" on the target with Edge AI base image -->
	<arg name="device_name"            default="/dev/video2"/>
	<!-- camera_mode: 'FHD' - 1920x1080, 'HD' - 1280x720, 'VGA' - 640x480 -->
	<arg name="camera_mode"            default="HD"/>
	<!-- frame rate: check the valid frame rates for camera_mode selected -->
	<arg name="frame_rate"             default="10"/>
	<!-- encoding: "yuv422" (default, output in YUV422:UYVY format), "bgr8" -->
	<arg name="encoding"               default="yuv422"/>
	<!-- topic_ns: topic namespace -->
	<arg name="topic_ns"               default="camera"/>

	<node pkg="mono_capture" exec="mono_capture" output="screen" required="true">
		<param name="camera_mode"            value="$(var camera_mode)"/>
		<param name="frame_rate"             value="$(var frame_rate)"/>
		<param name="frame_id"               value="camera_frame"/>
		<param name="device_name"            value="$(var device_name)"/>
		<param name="encoding"               value="$(var encoding)"/>
		<param name="image_topic"            value="$(var topic_ns)/image_raw"/>
		<param name="camera_info_topic"      value="$(var topic_ns)/camera_info"/>
		<param name="camera_info_yaml"       value="$(var model_str)_$(var camera_mode)_camera_info.yaml"/>
	</node>

</launch>
